numpy/distutils/command/build_ext.py

```diff
@@ -509,7 +509,11 @@ def _process_unlinkable_fobjects(self, objects, libraries,
 
         # Wrap unlinkable objects to a linkable one
         if unlinkable_fobjects:
-            fobjects = [os.path.relpath(obj) for obj in unlinkable_fobjects]
+            if os.name == 'nt':
+                # relpath fails if we are on a different drive
+                fobjects = [os.path.abspath(obj) for obj in unlinkable_fobjects]
+            else:
+                fobjects = [os.path.relpath(obj) for obj in unlinkable_fobjects]
```

Comment:
We don't actually test the alternate drive scenario since we are using `-n`, so I am hesitant to put in more complicated untested logic. At some point Python could decide to change the exception to a different error and we would never know. Perhaps we should just always use `abspath`?

Reply:
It's hard to say without knowing the original rationale for the relative paths.

I think it's incredibly unlikely that python changes the exception type in future, and if they do, I'd hope they'd just create a subclass of `ValueError`, like they did when they changed `IOError`